### PR TITLE
Updating MMA to use latest Optim

### PR DESCRIPTION
**Pull request by mohdibntarek**

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -3,13 +3,12 @@ function check_error(m, x0)
         throw(ArgumentError("initial variable must have same length as number of design variables"))
     end
 
-    for j in length(x0)
```
Comment (KristofferC) [inline src/utils.jl:6]: !!! :D

Comment (mohdibntarek) [reply src/utils.jl:6]: Happens!!

**Comment by KristofferC** [comment]
Tests passes on 0.6! Thanks for this. Is this good to go?

**Comment by mohdibntarek** [comment]
May need a bit of cleaning up. Give it a couple of days. 

I am using it for some topology optimization problem and Optim is giving a warning that line search failed which is killing my program, so something is not quite right yet, still looking into it.

**Comment by KristofferC** [comment]
I used it for:

https://www.youtube.com/watch?v=OO0DVTRMRuE
https://www.youtube.com/watch?v=KK3ggEkuPN8

:)

**Comment by mohdibntarek** [comment]
Cool, looks like SIMP with continuation?

**Comment by KristofferC** [comment]
I'm not sure what continuation is but the second has some smoothing operator applied to it to prevent checkerboarding.

**Comment by mohdibntarek** [comment]
I see, continuation just ramps up the power in SIMP to remove grey areas gradually.

**Comment by mohdibntarek** [comment]
Hey, can u try running the SIMP code u have with this MMA version? I am getting an error from Optim saying `ERROR: Search direction is not a direction of descent; this error may indicate that user-provided derivatives are inaccurate. (dphia = 0.000000; dphib = 0.000000)`. The error doesn't seem like it is from JuAFEM. Also the MMA code seems fine to me, so the only 2 possible sources of error are Optim and my SIMP code which I checked multiple times and see no problem with. If you are getting a similar error, then Optim's gradient descent is the source of the error for some reason, otherwise it's my SIMP code.

**Comment by mohdibntarek** [comment]
Nevermind, it's all good now! I will give MMA a final scan and tell you when it is ready to go.

**Comment by mohdibntarek** [comment]
I think it is fine now. But when I tried it with `@code_warntype`, there were some type instabilities. I could not figure out the reasons. So you may want to check it again for type stability or merge as is, whatever you want. I think I won't be retouching it for a while.

**Comment by KristofferC** [comment]
Thanks for updating this!

**Pull request merged**